test/e2e/case42_resource_selection_test.go

```diff
@@ -0,0 +1,211 @@
+		})
+
+		By("Verifying initial compliance message")
+		Eventually(func() interface{} {
+			managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+				policyName, testNamespace, true, defaultTimeoutSeconds)
+
+			return utils.GetStatusMessage(managedPlc)
+		}, defaultTimeoutSeconds, 1).Should(Equal(fmt.Sprintf(
+			"fakeapis [case42-1-e2e] found but not as specified in namespace %[1]s; "+
+				"fakeapis [case42-2-e2e] found but not as specified in namespace %[1]s", targetNs)))
+	})
+
+	It("should evaluate when a matching labeled resource is added", func(ctx SpecContext) {
+		By("Creating additional matching object case42-3-e2e")
+		utils.Kubectl("apply", "-n", targetNs, "-f", extraYaml)
+
+		Eventually(func() interface{} {
+			managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+				policyName, testNamespace, true, defaultTimeoutSeconds)
+
+			return utils.GetStatusMessage(managedPlc)
+		}, defaultTimeoutSeconds, 1).Should(HaveSuffix(
+			fmt.Sprintf("; fakeapis [case42-3-e2e] found but not as specified in namespace %s", targetNs)))
+	})
+
+	It("should not change when a non-matching resource is added", func(ctx SpecContext) {
+		By("Creating additional matching object case42-4-e2e")
+		utils.Kubectl("apply", "-n", targetNs, "-f", nomatchYaml)
+
+		Consistently(func() interface{} {
+			managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+				policyName, testNamespace, true, defaultTimeoutSeconds)
+
+			return utils.GetStatusMessage(managedPlc)
+		}, defaultTimeoutSeconds, 1).ShouldNot(ContainSubstring("case42-4-e2e"))
+	})
+
+	It("should evaluate when a resource is labeled to match", func() {
+		utils.Kubectl("label", "-n", targetNs, "fakeapi", "case42-4-e2e", "case42=case42-4-e2e")
+
+		Eventually(func() interface{} {
+			managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+				policyName, testNamespace, true, defaultTimeoutSeconds)
+
+			return utils.GetStatusMessage(managedPlc)
+		}, defaultTimeoutSeconds, 1).Should(HaveSuffix(
+			fmt.Sprintf("; fakeapis [case42-4-e2e] found but not as specified in namespace %s", targetNs)))
+	})
+
+	It("should evaluate when a matching resource label is removed", func() {
+		utils.Kubectl("label", "-n", targetNs, "fakeapi", "case42-3-e2e", "case42-")
+
+		Eventually(func() interface{} {
+			managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+				policyName, testNamespace, true, defaultTimeoutSeconds)
+
+			return utils.GetStatusMessage(managedPlc)
+		}, defaultTimeoutSeconds, 1).ShouldNot(ContainSubstring(
+			fmt.Sprintf("fakeapis [case42-3-e2e] found but not as specified in namespace %s", targetNs)))
+	})
+
+	It("should become compliant when enforced", func() {
+		utils.Kubectl("patch", "--namespace=managed", "configurationpolicy", policyName, "--type=json",
+			`--patch=[{"op":"replace","path":"/spec/remediationAction","value":"enforce"}]`,
+		)
```

Comment:
Isn't it better we add message check?  To check that only 1, 2,4 are applied 
```
 message: fakeapis [case42-1-e2e] found as specified in namespace case42-e2e-2;
        fakeapis [case42-2-e2e] found as specified in namespace case42-e2e-2; fakeapis
        [case42-4-e2e] found as specified in namespace case42-e2e-2
      reason: K8s `must have` object already exists
```

Reply:
Oh also another question, enforce with objectSelector will never create new resources right?

Reply:
> Isn't it better we add message check? To check that only 1, 2,4 are applied

That's fair since there isn't really another enforce test. I'll add it.

> enforce with objectSelector will never create new resources

Right, it can only discover and modify existing resources.

Reply:
Oh goodness! There's not even any assertion here! 😂 I'll _definitely_ add it--thanks for catching it!